test_communication/test/requester_py.py

```diff
@@ -17,55 +17,52 @@
 import sys
 
 
-def requester(service_name, number_of_cycles):
-    from test_msgs.service_fixtures import get_test_srv
+def requester(service_name):
     import rclpy
-
-    service_pkg = 'test_msgs'
-    module = importlib.import_module(service_pkg + '.srv')
-    srv_mod = getattr(module, service_name)
-
+    from test_msgs.service_fixtures import get_test_srv
     rclpy.init(args=[])
 
-    node = rclpy.create_node('requester')
+    try:
+        # Import the service
+        service_pkg = 'test_msgs'
+        module = importlib.import_module(service_pkg + '.srv')
+        srv_mod = getattr(module, service_name)
 
-    srv_fixtures = get_test_srv(service_name)
+        srv_fixtures = get_test_srv(service_name)
+        service_name = 'test_service_' + service_name
 
-    client = node.create_client(srv_mod, 'test_service_' + service_name)
+        node = rclpy.create_node('requester')
+        try:
+            # wait for the service to be available
+            client = node.create_client(srv_mod, service_name)
+            tries = 15
+            while rclpy.ok() and not client.wait_for_service(timeout_sec=1.0) and tries > 0:
+                print('service not available, waiting again...')
+                tries -= 1
+            assert tries > 0
 
-    spin_count = 1
-    received_replies = []
-    print('requester: beginning loop')
-    while rclpy.ok() and spin_count < number_of_cycles:
-        for req, resp in srv_fixtures:
-            client.call(req)
-            client.wait_for_future()
-            assert repr(client.response) == repr(resp), \
-                'received unexpected response %r\n\nwas expecting %r' % (client.response, resp)
-            print('received reply #%d of %d' %
-                  (srv_fixtures.index([req, resp]) + 1, len(srv_fixtures)))
-            received_replies.append(resp)
-            spin_count += 1
-            print('spin_count: ' + str(spin_count))
-        break
-    node.destroy_node()
-    rclpy.shutdown()
-    assert len(received_replies) == len(srv_fixtures), \
-        'Should have received %d responses from replier' % len(srv_fixtures)
-    print('everything went well !')
+            print('requester: beginning request')
+            # Make one call to that service
+            for req, resp in srv_fixtures:
+                client.call(req)
+                client.wait_for_future()
```

Comment:
if this doesn't return in a timely manner we still have potential for timing out without cleanup. That was the problem before this PR and is out of scope but should be ticketed so that we address it soon